File: src/main/java/BluetoothService.java
import android.bluetooth.BluetoothSocket;

public class BluetoothService {
    private static BluetoothSocket socket;

    public static synchronized void setBluetoothSocket(BluetoothSocket bluetoothSocket) {
        BluetoothService.socket = bluetoothSocket;
    }

    public static synchronized BluetoothSocket getBluetoothSocket() {
        return BluetoothService.socket;
    }

    // Add methods to manage the connection, read, write, and close operations.
}

File: src/test/java/BluetoothService.kt
package com.example.ssjbeta2

import android.bluetooth.BluetoothSocket

object BluetoothService {
    private var bluetoothSocket: BluetoothSocket? = null

    @Synchronized
    fun setBluetoothSocket(socket: BluetoothSocket?) {
        bluetoothSocket = socket
    }

    @Synchronized
    fun getBluetoothSocket(): BluetoothSocket? {
        return bluetoothSocket
    }

    // Add other Bluetooth management methods as needed.
}
